### ISSUE TITLE
Do we need `IAvoidDuplicateCrossCuttingConcerns`

### ISSUE THREAD

**Issue by ahmetfarukulu** [abpframework/abp#15538]
### Description

As far as I understand it controllers/pages [don't use](https://github.com/abpframework/abp/blob/6c02fb826c91e8dc03c222d73f7a76efbda9d631/framework/src/Volo.Abp.AspNetCore.Mvc/Volo/Abp/AspNetCore/Mvc/AbpAspNetCoreMvcModule.cs#L61) interceptions for [performance issue](https://github.com/castleproject/Core/issues/486) but we're adding some unnecessary check

### Analysis
As you notice [apply](https://github.com/abpframework/abp/blob/6c02fb826c91e8dc03c222d73f7a76efbda9d631/framework/src/Volo.Abp.Core/Volo/Abp/Aspects/AbpCrossCuttingConcerns.cs#L60) method take object as parameter to avoid duplicate CCC and we give controller/page for that also we only use with filters but filters object(Controller/Page) never intercept by AbpInterception

 - [AbpFeaturePageFilter](https://github.com/abpframework/abp/blob/6c02fb826c91e8dc03c222d73f7a76efbda9d631/framework/src/Volo.Abp.AspNetCore.Mvc/Volo/Abp/AspNetCore/Mvc/Features/AbpFeaturePageFilter.cs#L27) and [AbpFeatureActionFilter](https://github.com/abpframework/abp/blob/6c02fb826c91e8dc03c222d73f7a76efbda9d631/framework/src/Volo.Abp.AspNetCore.Mvc/Volo/Abp/AspNetCore/Mvc/Features/AbpFeatureActionFilter.cs#L22) avoid duplicate CCC and cheking from [FeatureInterceptor](https://github.com/abpframework/abp/blob/6c02fb826c91e8dc03c222d73f7a76efbda9d631/framework/src/Volo.Abp.Features/Volo/Abp/Features/FeatureInterceptor.cs#L21)
 - [GlobalFeaturePageFilter](https://github.com/abpframework/abp/blob/6c02fb826c91e8dc03c222d73f7a76efbda9d631/framework/src/Volo.Abp.AspNetCore.Mvc/Volo/Abp/AspNetCore/Mvc/GlobalFeatures/GlobalFeaturePageFilter.cs#L37) and [GlobalFeatureActionFilter](https://github.com/abpframework/abp/blob/6c02fb826c91e8dc03c222d73f7a76efbda9d631/framework/src/Volo.Abp.AspNetCore.Mvc/Volo/Abp/AspNetCore/Mvc/GlobalFeatures/GlobalFeatureActionFilter.cs#L32) avoid duplicate CCC and cheking from [GlobalFeatureInterceptor](https://github.com/abpframework/abp/blob/6c02fb826c91e8dc03c222d73f7a76efbda9d631/framework/src/Volo.Abp.GlobalFeatures/Volo/Abp/GlobalFeatures/GlobalFeatureInterceptor.cs#L12)
 - [AbpAuditPageFilter](https://github.com/abpframework/abp/blob/6c02fb826c91e8dc03c222d73f7a76efbda9d631/framework/src/Volo.Abp.AspNetCore.Mvc/Volo/Abp/AspNetCore/Mvc/Auditing/AbpAuditPageFilter.cs#L28) and [AbpAuditActionFilter](https://github.com/abpframework/abp/blob/6c02fb826c91e8dc03c222d73f7a76efbda9d631/framework/src/Volo.Abp.AspNetCore.Mvc/Volo/Abp/AspNetCore/Mvc/Auditing/AbpAuditActionFilter.cs#L23) avoid duplicate CCC and cheking from [AuditingInterceptor](https://github.com/abpframework/abp/blob/6c02fb826c91e8dc03c222d73f7a76efbda9d631/framework/src/Volo.Abp.Auditing/Volo/Abp/Auditing/AuditingInterceptor.cs#L60)

**Comment by maliming** [comment]
hi

We will try to ignore the `Interceptors` if there is a filter applied.

https://github.com/abpframework/abp/search?l=C%23&q=IsApplied

**Comment by ahmetfarukulu** [comment]
Hi, 

We only [apply](https://github.com/search?q=repo%3Aabpframework%2Fabp+AbpCrossCuttingConcerns.Applying&type=code) for action/page filters and apply object is controller/page but we don't use interception for this types

In this [issue](https://github.com/abpframework/abp/issues/8108) reason for apply to not check twice but we can use AsyncLocal

```csharp
public class NotifyOnExceptionInterceptor : AbpInterceptor, ITransientDependency
{
    private static readonly AsyncLocal<bool> IsExecute = new AsyncLocal<bool>();

    public NotifyOnExceptionInterceptor()
    {

    }

    public override async Task InterceptAsync(IAbpMethodInvocation invocation)
    {
        if (IsExecute.Value)
        {
            await invocation.ProceedAsync();
            return;
        }

        try
        {
            IsExecute.Value = true;
            await invocation.ProceedAsync();
        }
        catch (Exception e)
        {
            //Log exception
            throw;
        }
    }
}
```

**Comment by maliming** [comment]
`private static readonly AsyncLocal<bool> IsExecute = new AsyncLocal<bool>();`

I think this won't work. Did you try that?

**Comment by ahmetfarukulu** [comment]
It works important thing is interception start with `Method1Async` with this way every sub methods include the same context

![image](https://user-images.githubusercontent.com/35758317/216270922-b5e8be66-ba55-4856-b085-f65d8e5d4f48.png)

**Comment by maliming** [comment]
Does it also work for multiple http requests?

**Comment by ahmetfarukulu** [comment]
Yes every http requests different scope

**Comment by maliming** [comment]
hi

Can you open a pr? then I will test it. Thanks

**Comment by ahmetfarukulu** [comment]
Hi @maliming

Filter avoid object and interception target object is different that's why `IAvoidDuplicateCrossCuttingConcerns` not work as expected. It check twice therefore these [usings](https://github.com/abpframework/abp/blob/6c02fb826c91e8dc03c222d73f7a76efbda9d631/framework/src/Volo.Abp.AspNetCore.Mvc/Volo/Abp/AspNetCore/Mvc/Features/AbpFeatureActionFilter.cs#L22) are unnecessary.

I think there is some confusion here [BlogAdminAppService](https://github.com/abpframework/abp/blob/bb1d48209480e7ca66bde7eb50a1c73c505eb41d/modules/cms-kit/src/Volo.CmsKit.Admin.Application/Volo/CmsKit/Admin/Blogs/BlogAdminAppService.cs#L19) and [BlogAdminController](https://github.com/abpframework/abp/blob/bb1d48209480e7ca66bde7eb50a1c73c505eb41d/modules/cms-kit/src/Volo.CmsKit.Admin.HttpApi/Volo/CmsKit/Admin/Blogs/BlogAdminController.cs#L21) use same attributes(Authorize, RequiresFeature) in that case it check twice first action filter check after interception check(interception target object use BlogAdminAppService and action filter avoid object use BlogAdminController)

If we want to check once then it must be like [TenantAppService](https://github.com/abpframework/abp/blob/bb1d48209480e7ca66bde7eb50a1c73c505eb41d/modules/tenant-management/src/Volo.Abp.TenantManagement.Application/Volo/Abp/TenantManagement/TenantAppService.cs#L14) and [TenantController](https://github.com/abpframework/abp/blob/bb1d48209480e7ca66bde7eb50a1c73c505eb41d/modules/tenant-management/src/Volo.Abp.TenantManagement.HttpApi/Volo/Abp/TenantManagement/TenantController.cs#L13)(Authorize check only TenantAppService)

In conclusion i think we can remove `IAvoidDuplicateCrossCuttingConcerns` and related things. Ofcourse this is not a complete solution it still check twice for prevent this we can check only on appservice side
